src/v/kafka/server/handlers/txn_offset_commit.cc

```diff
@@ -87,6 +98,16 @@ ss::future<response_ptr> txn_offset_commit_handler::handle(
     if (unlikely(ctx.recovery_mode_enabled())) {
         return ctx.respond(
           txn_offset_commit_response{request, error_code::policy_violation});
+    } else if (!ctx.authorized(
```

Comment:
same question here as above

Reply:
In this case, we're unauthorized on the transactional ID itself, so yeah, fail immediately.